src/main/resources/com/microsoft/jenkins/azuread/AzureSecurityRealm/config.jelly

```diff
@@ -6,34 +6,47 @@
 
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     <f:block>
-        <f:entry title="Client ID"  field="clientId" help="/plugin/azure-ad/help/help-client-id.html" >
+        <f:entry title="Client ID"  field="clientId" >
             <f:textbox />
         </f:entry>
 
-        <f:entry title="Client Secret" field="clientSecret" help="/plugin/azure-ad/help/help-client-secret.html">
-            <f:password />
+        <f:entry title="Authentication Type">
+            <f:radioBlock name="credentialType" value="Secret" title="Client Secret" help="${null}"
+                          checked="${(instance.credentialType == null) ? 'true' : (instance.credentialType == 'Secret')}" inline="true">
+                <f:entry title="Secret" field="clientSecret">
+                    <f:password/>
+                </f:entry>
+            </f:radioBlock>
+            <f:radioBlock name="credentialType" value="Certificate" title="Client Certificate"
+                          help="${null}"
+                          checked="${instance.credentialType == 'Certificate'}" inline="true">
+                <f:entry title="Certificate" field="clientCertificate">
```

Comment:
Dropped `Client` so that you didn't have 2 labels following each other with the same text that didn't look good